### PR TITLE
Add REST API Client Code Generator section to install.md

### PR DESCRIPTION
**Pull request by christianhelle**
The changes here are based off a [suggestion by](https://github.com/microsoft/kiota/issues/3005#issuecomment-1649746008) @baywet in [this discussion](https://github.com/microsoft/kiota/issues/3005) to add information about [REST API Client Code Generator](https://github.com/christianhelle/apiclientcodegen) as a community provided tooling option for Kiota.

I added the following to `install.md`

## REST API Client Code Generator

> [!IMPORTANT]
> The REST API Client Code Generator extension for Visual Studio is maintained and distributed by the community and is not an official Microsoft Visual Studio extension. Microsoft makes no warranties, express or implied, with respect to the extension or its use. Use of this extension is at your own risk. Microsoft shall not be liable for any damages arising out of or in connection with the use of this extension.

[REST API Client Code Generator](https://github.com/christianhelle/apiclientcodegen) is a collection of Visual Studio C# custom tool code generators for OpenAPI specifications. This extension will install Kiota on-demand and will also add the required NuGet packages to build the generated code to the project. The generated code is created as a "code-behind" file to the OpenAPI specifications file in the .NET project. This extension offers same-day releases for new Kiota versions, but this requires updating the extension, which can be configured to be automatically.

**Installation:**

1. From Visual Studio (Windows), using the [Manage Extensions](https://learn.microsoft.com/en-us/visualstudio/ide/finding-and-using-visual-studio-extensions) dialog box (Tools -> Manage Extensions), search for extension  called [REST API Client Code Generator](https://marketplace.visualstudio.com/items?itemName=ChristianResmaHelle.ApiClientCodeGenerator2022). This extension is available for [Visual Studio 2022 for AMD64 and ARM64](https://marketplace.visualstudio.com/items?itemName=ChristianResmaHelle.ApiClientCodeGenerator2022) and [Visual Studio 2019](https://marketplace.visualstudio.com/items?itemName=ChristianResmaHelle.ApiClientCodeGenerator)

   For Visual Studio for Mac, follow [these instructions](https://github.com/christianhelle/apiclientcodegen/blob/master/docs/VisualStudioForMac.md#installation)

2. Click on the **Download** button. The extension will be installed after closing Visual Studio so you will need to do a restart

**Usage:**

To generate code using Kiota, Right click on a project or location under a project and select **Add -> New REST API Client -> Generate with Microsoft Kiota**

add-new

This will prompt you for a URL to download the OpenAPI specifications file.

If the operation worked successfully then you should see a code-behind file for the OpenAPI specifications. Once you have generated code, all changes to the OpenAPI specifications file will trigger a re-generation of the code-behind file via the **KiotaCodeGenerator** custom tool

custom-tool-experience

If you already have a OpenAPI specifications file in the project, then you can also generate code using Kiota by right clicking on the file then selecting **REST API Client Generator -> Generate with Microsoft Kiota**

generate-from-existing

### CODE REVIEW

**Comment by christianhelle** [comment]
@microsoft-github-policy-service agree

**Comment by christianhelle** [comment]
Sorry but I don't know how to resolve

**Line 110, Column 44: [Suggestion: docs-link-absolute - [See documentation](https://review.learn.microsoft.com/help/contribute/validation-ref/docs-link-absolute?branch=main)]** `Absolute link 'https://learn.microsoft.com/en-us/visualstudio/ide/finding-and-using-visual-studio-extensions' will be broken in isolated environments. Replace with a relative link.`

Ay hints?

File: OpenAPI/kiota/install.md [hunk 1]
```diff
@@ -16,7 +16,10 @@ Kiota can be accessed in the following ways.
 - [Install as .NET tool](#install-as-net-tool)
 - [Build from source](#build-from-source)
 - [Install the Visual Studio Code extension (preview)](#install-the-visual-studio-code-extension)
-- [Install with asdf (unofficial)](#install-with-asdf)
+
+Unofficial Community provided options:
```
Comment (baywet) [inline OpenAPI/kiota/install.md:20]:
```suggestion
Community provided options:
```

File: OpenAPI/kiota/install.md [hunk 2]
```diff
@@ -94,3 +97,34 @@ For details on running Kiota, see [Using the Kiota tool](using.md).
...
+
+This will prompt you for a URL to download the OpenAPI specifications file.
+
+If the operation worked successfully then you should see a code-behind file for the OpenAPI specifications. Once you have generated code, all changes to the OpenAPI specifications file will trigger a re-generation of the code-behind file via the **KiotaCodeGenerator** custom tool
+
+![Custom Tool Experience](./images/community/rest-api-client-generator/custom-tool-experience.png)
+
+If you already have a OpenAPI specifications file in the project, then you can also generate code using Kiota by right clicking on the file then selecting **REST API Client Generator -> Generate with Microsoft Kiota**
+
+![Genrate code using Kiota](./images/community/rest-api-client-generator/generate-from-existing.png)
```
Comment (baywet) [inline OpenAPI/kiota/install.md:130]:
```suggestion
![Generate code using Kiota](./images/community/rest-api-client-generator/generate-from-existing.png)
```

Comment (baywet) [inline OpenAPI/kiota/install.md:130]: @jasonjoh what do you think about adding the usage here? It'd be the only installation media we do it for.

**Review by baywet — CHANGES_REQUESTED** [review]

**Comment by jasonjoh** [comment]
> Sorry but I don't know how to resolve
> 
> **Line 110, Column 44: [Suggestion: docs-link-absolute - [See documentation](https://review.learn.microsoft.com/help/contribute/validation-ref/docs-link-absolute?branch=main)]** `Absolute link 'https://learn.microsoft.com/en-us/visualstudio/ide/finding-and-using-visual-studio-extensions' will be broken in isolated environments. Replace with a relative link.`
> 
> Ay hints?

@christianhelle remove the `https://learn.microsoft.com/en-us` bit from the URL to make it relative.

File: OpenAPI/kiota/install.md (hunk 2 above)
Comment (jasonjoh) [reply OpenAPI/kiota/install.md]: I see your point. We should definitely just link to the extension's usage documentation rather than list it out here.

Comment (baywet) [reply OpenAPI/kiota/install.md]: Thanks for confirming. @christianhelle do you mind making that update as well as the one for the docs link please? after this we should be good to merge.

Comment (christianhelle) [reply OpenAPI/kiota/install.md]: @baywet will do

Comment (christianhelle) [reply OpenAPI/kiota/install.md]: Done

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

File: OpenAPI/kiota/install.md [hunk 3]
```diff
@@ -16,7 +16,10 @@ Kiota can be accessed in the following ways.
 - [Install as .NET tool](#install-as-net-tool)
 - [Build from source](#build-from-source)
 - [Install the Visual Studio Code extension (preview)](#install-the-visual-studio-code-extension)
-- [Install with asdf (unofficial)](#install-with-asdf)
+
+Community provided options:
+- [Install with asdf](#install-with-asdf)
```
Comment (baywet) [inline OpenAPI/kiota/install.md:21]:
```suggestion

- [Install with asdf](#install-with-asdf)
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**